frontend/packages/dev-console/src/components/monitoring/metrics/__tests__/MetricsQueryInput.spec.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { Button } from '@patternfly/react-core';
+import { Dropdown } from '@console/internal/components/utils';
+import { QueryInput } from '@console/internal/components/monitoring/metrics';
+import { MetricsQueryInput } from '../MetricsQueryInput';
+
+describe('Metrics Query Input', () => {
+  let props: React.ComponentProps<typeof MetricsQueryInput>;
+  beforeEach(() => {
+    props = {
+      namespace: 'my-app',
+      patchQuery: jest.fn(),
+      runQueries: jest.fn(),
+    };
+  });
+
+  it('should render Dropdown with default title', () => {
+    const wrapper = shallow(<MetricsQueryInput {...props} />);
+    expect(wrapper.find(Dropdown)).toHaveLength(1);
+    expect(wrapper.find(Dropdown).props().title).toEqual('Select Query');
+  });
+
+  it('should render Button with text "Show PromQL" and not render QueryInput', () => {
```

Comment:
As commented above by default "Show PromQL" button need to be hidden or at least disabled. May be it's better to assert this state as well.

Reply:
`Show PromQL button` is there in the mockup